vllm/v1/core/sched/scheduler.py

```diff
@@ -1483,29 +1510,78 @@ def _handle_invalid_blocks(self, invalid_block_ids: set[int]) -> set[str]:
             )
         )
 
-        total_requests_to_reschedule += len(async_affected_req_ids)
-        total_tokens_to_reschedule += num_tokens_to_reschedule
-
-        # Mark requests with async KV load failures; they will be rescheduled
-        # once loading completes.
-        self.failed_recving_kv_req_ids |= async_affected_req_ids
+        total_requests_to_reschedule = len(async_affected_req_ids)
+        total_tokens_to_reschedule = num_tokens_to_reschedule
 
-        # --- Handle sync KV loads (running requests) ---
+        # Handle sync KV loads (running requests)
         sync_affected_req_ids, num_tokens_to_reschedule = (
             self._update_requests_with_invalid_blocks(self.running, invalid_block_ids)
         )
 
         total_requests_to_reschedule += len(sync_affected_req_ids)
         total_tokens_to_reschedule += num_tokens_to_reschedule
 
-        if total_requests_to_reschedule:
+        if not total_requests_to_reschedule:
+            return set()
+
+        # --- Take action based on policy ---
+        if abort:
+            logger.warning(
+                "Aborting %d request(s) due to KV load failure "
+                "(retry_policy=abort, %d tokens affected).",
+                total_requests_to_reschedule,
+                total_tokens_to_reschedule,
+            )
+            all_affected_req_ids = async_affected_req_ids | sync_affected_req_ids
+
+            # track which requests to remove from queues
+            stopped_running_reqs: set[Request] = set()
+            stopped_waiting_reqs: list[Request] = []
+
+            # create EngineOutputs for each aborted request
+            for req_id in all_affected_req_ids:
+                request = self.requests.get(req_id)
+                if request is None:
+                    continue
+
+                # track which queue the request was in
+                if request.status == RequestStatus.RUNNING:
+                    stopped_running_reqs.add(request)
+                else:
+                    stopped_waiting_reqs.append(request)
+
+                # abort and free the request
+                request.status = RequestStatus.FINISHED_ERROR
+                kv_transfer_params = self._free_request(request)
```

Comment:
Any reason not to use `finish_requests()` here? At a glance, it would replace much of this logic?